registrations/api.py

```diff
@@ -266,25 +265,14 @@ def create(self, request, *args, **kwargs):
 
         return Response(data, status=status.HTTP_201_CREATED)
 
+    @transaction.atomic
     def destroy(self, request, *args, **kwargs):
         instance = self.get_object()
-        registration = instance.registration
-
-        instance.send_notification(SignUpNotificationType.CANCELLATION)
-        response = super().destroy(request, *args, **kwargs)
-
-        # Move first signup from waitlist to attending list
-        waitlisted = registration.signups.filter(
-            attendee_status=SignUp.AttendeeStatus.WAITING_LIST,
-        ).order_by("id")
-        if len(waitlisted) > 0:
-            first_on_list = waitlisted[0]
-            first_on_list.attendee_status = SignUp.AttendeeStatus.ATTENDING
-            first_on_list.save()
-            first_on_list.send_notification(
-                SignUpNotificationType.TRANSFERRED_AS_PARTICIPANT
-            )
-        return response
+
+        instance._individually_deleted = True
+        self.perform_destroy(instance)
+
+        return Response(status=status.HTTP_204_NO_CONTENT)
```

Comment:
You could also remove `destroy` and replace it with:

```python
    @transaction.atomic
    def perform_destroy(self, instance):
        instance._individually_deleted = True
        instance.delete()
```

Reply:
Changed as suggested.